### PR TITLE
Expand contents of SecurityContextConstraints

### PR DESCRIPTION
**Pull request by yossisegev**
Support extra parameters of SecurityContextConstraints resource.

### CODE REVIEW

File: ocp_resources/security_context_constraints.py [hunk 1]
```diff
@@ -3,7 +3,196 @@
 
 class SecurityContextConstraints(Resource):
     """
-    Security Context Constraints object.
+    Security Context Constraints object. API reference:
+    https://docs.openshift.com/container-platform/4.13/rest_api/security_apis/securitycontextconstraints-security-openshift-io-v1.html
+
+    Args:
```
Comment (rnetser) [inline ocp_resources/security_context_constraints.py:9]: 1. should be under __init__
2. do not repeat the api docs
3. defaults by the api should not appear here

File: ocp_resources/security_context_constraints.py [hunk 2]
```diff
@@ -3,7 +3,196 @@
...
+        self.seccomp_profiles = seccomp_profiles
+        self.supplemental_groups = supplemental_groups
+        self.users = users
+        self.volumes = volumes
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.update()
+            manifest_spec = self.res
```
Comment (rnetser) [inline ocp_resources/security_context_constraints.py:117]: no need to save

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/security_context_constraints.py (hunk 1 above)
Comment (yossisegev) [reply ocp_resources/security_context_constraints.py:9]: Done

File: ocp_resources/security_context_constraints.py (hunk 2 above)
Comment (yossisegev) [reply ocp_resources/security_context_constraints.py:117]: Done

**Review by myakove — APPROVED** [review]

File: ocp_resources/security_context_constraints.py [hunk 3]
```diff
@@ -2,8 +2,174 @@
...
+        users=None,
+        volumes=None,
+        **kwargs,
+    ):
+        """
+        Security Context Constraints object. API reference:
+        https://docs.openshift.com/container-platform/4.13/rest_api/security_apis/securitycontextconstraints-security-openshift-io-v1.html
+
+        Args:
+            allow_host_dir_volume_plugin (bool): Determines if the policy allow containers to use the HostDir volume
```
Comment (rnetser) [inline ocp_resources/security_context_constraints.py:40]: please add defaults, e.g `allow_host_dir_volume_plugin (bool, default: False)`

File: ocp_resources/security_context_constraints.py [hunk 4]
```diff
@@ -2,8 +2,174 @@
...
+            if self.forbidden_sysctls:
+                self.res["forbiddenSysctls"] = self.forbidden_sysctls
+
+            if self.fs_group:
+                self.res["fsGroup"] = self.fs_group
+
+            if self.groups:
+                self.res["groups"] = self.groups
+
+            if self.priority:
```
Comment (rnetser) [inline ocp_resources/security_context_constraints.py:150]: `priority` default is `0`
is `0` a valid option?if so, the code below will not be reached

**Review by rnetser — CHANGES_REQUESTED** [review]

**Pull request closed without merging**